pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -364,6 +376,10 @@ func (bfs *Service) streamBlockOIDs(rc io.ReadCloser, startIndex, skip int) erro
 		case bfs.oidsCh <- indexedOID{Index: startIndex*int(bfs.cfg.IndexFileSize) + oidsProcessed, OID: oidBlock}:
 		}
 
+		if bfs.stopAt > 0 && startIndex*int(bfs.cfg.IndexFileSize)+oidsProcessed == int(bfs.stopAt) {
```

Comment:
`startIndex*int(bfs.cfg.IndexFileSize)+oidsProcessed` is duplicated. Allocate variable, reuse it. Cast to uint32 instead of int.